js/app.js

```diff
@@ -0,0 +1,548 @@
+outputTypes[OUTPUT_CSV] = { name: OUTPUT_CSV };
+outputTypes[OUTPUT_YAML] = { name: OUTPUT_YAML };
+
+var GeneratorApp = React.createClass({
+    getInitialState: function() {
+        return {
+            columns: this.props.columns,
+            settings: this.props.settings,
+            outputTypes: this.props.outputTypes,
+            selectedOutputType: OUTPUT_MYSQL,
+            output: ''
+        };
+    },
+    componentWillMount: function() {
+        this.getOutput();
+    },
+    // handles changes for all the checkboxes and radio buttons
+    toggleCheck: function(objectKey, type, data) {
+        // radio buttons are handled different than checkboxes
+        if (type === 'outputTypes') {
+            var settings = this.state.settings;
+
+            // based on the selected, disable or enable 'languagelookup' setting
+            for (var key in settings) {
+                if (settings.hasOwnProperty(key) && settings[key].name === 'languagelookup') {
+                    if (settings[key].supportedOutputs.indexOf(data[objectKey].name) === -1) {
+                        settings[key].checked = false;
+                        settings[key].disabled = true;
+                    } else {
+                        settings[key].disabled = false;
+                    }
+                }
+            }
+
+            this.setState({selectedOutputType: data[objectKey].name});
+        } else if (type === 'settings') {
+            if (data[objectKey].name === 'languagelookup' && !data[objectKey].checked) {
+                var columns = this.state.columns;
+
+                columns['languages'].checked = true;
+            }
+
+            data[objectKey].checked = !data[objectKey].checked;
+        } else {
+            data[objectKey].checked = !data[objectKey].checked;
+        }
+
+        this.forceUpdate();
+
+        // TODO: do I need this? How forceUpdate does updates based on references?
+//        this.setState({type: data});
+    },
+    getOutput: function(e) {
+        if (e) {
+            e.preventDefault();
+        }
+
+        // TODO: show loader on button
+
+        var source = 'http://api.geonames.org/countryInfoJSON?username=dperic';
+        var columns = this.state.columns;
+        var settings = this.state.settings;
+        var selectedOutputType = this.state.selectedOutputType;
+
+        $.getJSON(source)
+            .done(function(data) {
+                var geonamesData = data.geonames;
+
+                this.setState({'output': generateOutput(selectedOutputType, columns, settings, geonamesData) });
+        }.bind(this));
+    },
+    render: function() {
+        var self = this;
+        var columns = Object.keys(this.state.columns).map(function(key, index) {
+            var column = self.state.columns[key];
+            return (
+                <Column
+                    key={index + column.name}
+                    objectKey={key}
+                    data={self.state.columns}
```

Comment:
`data` is really generic name. I believe you can think of something more specific ;)